tests/evaluate/test_default_evaluator.py

```diff
@@ -2927,3 +2927,78 @@ def test_evaluate_with_latency():
         "token_count",
     }
     assert all(isinstance(grade, float) for grade in logged_data["latency"])
+
+
+def test_evaluate_with_latency_static_dataset():
+    with mlflow.start_run() as run:
+        mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a", "b"]
+        )
+        data = pd.DataFrame(
+            {
+                "text": ["foo", "bar"],
+                "model_output": ["FOO", "BAR"],
+            }
+        )
+        results = mlflow.evaluate(
+            data=data,
+            model_type="text",
+            evaluators="default",
+            predictions="model_output",
+            extra_metrics=[mlflow.metrics.latency],
+        )
+
+    client = mlflow.MlflowClient()
+    artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+    assert "eval_results_table.json" in artifacts
+    logged_data = pd.DataFrame(**results.artifacts["eval_results_table"].content)
+    assert set(logged_data.columns.tolist()) == {
+        "text",
+        "outputs",
+        "toxicity/v1/score",
+        "flesch_kincaid_grade_level/v1/score",
+        "ari_grade_level/v1/score",
+        "perplexity/v1/score",
+        "latency",
+        "token_count",
+    }
+    assert all(isinstance(grade, float) for grade in logged_data["latency"])
+    assert all(grade == 0.0 for grade in logged_data["latency"])
+
+
+def test_default_metrics_as_custom_metrics_static_dataset():
```

Comment:
@sunishsheth2009 I'm not sure what is the purpose here, but I just made a counterpart of https://github.com/mlflow/mlflow/blob/a4db4ee826765f0365944f83124af8840c72f4d8/tests/evaluate/test_default_evaluator.py#L2888 with `model=None`. Could you help me understand why do we need `"predicted_column": "answer",`?

Reply:
So if model.predict() returns back multiple outputs. Users need to say which of the column out of the multiple column is the output column. Example langchain model would return `source_docs, result`, etc back. predicted_column here would be `result` 

So instead of adding a new test. Can we update the existing the test: https://github.com/mlflow/mlflow/blob/a4db4ee826765f0365944f83124af8840c72f4d8/tests/evaluate/test_default_evaluator.py#L2888 to not use eval_config and rather use your prediction arg?

Reply:
So you are suggesting extending the semantics of the `predictions` parameter when `model is not None`. Currently, the `predictions` parameter is only allowed to use `model=None`. I'm worrying that such a semantics extension may introduce more confusion.
@dbczumar Any thoughts?
also cc @chenmoneygithub @prithvikannan @bbqiu @annzhang-db for more inputs.

Reply:
Is removing the option to have a predicted column in the evaluator config an option?

I agree that having both as options would be confusing, but I do like the idea of making the `predictions` parameter *always* correspond to the "result" column (whether or not we are evaluating a static dataset)

Reply:
Yes that is what I am suggesting. We should remove predicted column in the evaluator config and use `predictions` which is a top level arg as the only way users can provide the column_name for prediction.

Reply:
```python
# case 1, evaluating with a model, the model is an identity model, it takes 2 input columns "input_col_1" and "input_col_2", and returns the input.
data = pd.DataFrame(
    {
        "input_col_1": ["words random", "This is a sentence."],
        "input_col_2": ["words random", "This is a sentence."],
        "label": ["words random", "This is a sentence."],
    }
)
results = evaluate(
    model=model_info.model_uri,
    data=data,
    targets="label",
    predictions="input_col_1",   # the original test says this, but I don't quite understand why do we specify an input column name here
    ...
)
# case 2, evaluating with a static dataset. We cannot support multi-output with a single `predictions` column name.
data = pd.DataFrame(
    {
        "input_col_1": ["words random", "This is a sentence."],
        "input_col_2": ["words random", "This is a sentence."],
        "label": ["words random", "This is a sentence."],
        "output_col_1": ["words random", "This is a sentence."],
        "output_col_2": ["words random", "This is a sentence."],
    }
)
results = evaluate(
    data=data,
    targets="label",
    predictions="output_col_1",  # If we specify this, then "input_col_1", "input_col_2", "output_col_2" will be treated as input columns.
    ...
)

```

Reply:
I understand the confusion her now.
In case 1:
```
# case 1, evaluating with a model, the model is an identity model (What is the output column names? Could they be the same as the input columns? Is this allowed?), it takes 2 input columns "input_col_1" and "input_col_2", and returns the input.
data = pd.DataFrame(
    {
        "input_col_1": ["words random", "This is a sentence."],
        "input_col_2": ["words random", "This is a sentence."],
        "label": ["words random", "This is a sentence."],
    }
)

def model(inputs):
   return {
       source_docs: '123',
       output: '543'
   }

results = evaluate(
    model=model,
    data=data,
    targets="label",
    predictions="output", 
    ...
    evalutor_config={
        col_mapping: {
            'context': 'source_docs'
        }
    }
)
```

So we need a way for users to mention which column is the output column from multiple output columns returned back the model. Does that makes sense?

Reply:
This means we don't allow the model to be an identity model, since the column name cannot be the same. Correct?
Also, for case 2, the behavior of feature_names=None needs to be changed:
https://github.com/mlflow/mlflow/blob/cf6b0e4b86597b33684d42ab6b7a4d6e139ba64c/mlflow/models/evaluation/base.py#L1454-L1461 

Needs to change the last sentence to:
```
If ``None``, then all columns except the label column, the predictions column, and all columns
in evalutor_config[evaluator_name]["col_mapping"].values() are regarded as feature columns.
```
which sounds complicated (but doable)... Is this what we want?

Reply:
>Needs to change the last sentence to:
If ``None``, then all columns except the label column, the predictions column, and all columns
in evalutor_config[evaluator_name]["col_mapping"].values() are regarded as feature columns.


I think `evalutor_config[evaluator_name]["col_mapping"].values()` are also feature columns. They could just be named differently if users really want to name them differently. If not, thats ok. 



> Also, for case 2, the behavior of feature_names=None needs to be changed:

For case 2 nothing changes.

Reply:
Basically we just need to replace these to lines:
https://github.com/mlflow/mlflow/blob/cf6b0e4b86597b33684d42ab6b7a4d6e139ba64c/mlflow/models/evaluation/default_evaluator.py#L1361

and 
https://github.com/mlflow/mlflow/blob/cf6b0e4b86597b33684d42ab6b7a4d6e139ba64c/mlflow/models/evaluation/default_evaluator.py#L1162

to read from prediction column instead of reading from evaluator_config.
Does that help clear the confusion? 

Also does the overall API make sense?

Reply:
OK, so for case 2, no code needs to be changed, but just the interpretation of columns in `data` changed.
* When evaluating using a model, `source` is an output column that is not used in the evaluation.
* When evaluating using a static dataset, `source` is an input column, which is also not used in the evaluation, but could potentially be used in the future (when we implement the evaluation for retriever and LLM together).

I can make the change, thanks @sunishsheth2009 !